src/wp-admin/setup-config.php

```diff
@@ -426,33 +426,56 @@ function setup_config_display_header( $body_classes = array() ) {
 })();
 </script>
 			<?php
-	else :
-		/*
-		 * If this file doesn't exist, then we are using the wp-config-sample.php
-		 * file one level up, which is for the develop repo.
-		 */
-		if ( file_exists( ABSPATH . 'wp-config-sample.php' ) ) {
-			$path_to_wp_config = ABSPATH . 'wp-config.php';
-		} else {
-			$path_to_wp_config = dirname( ABSPATH ) . '/wp-config.php';
-		}
+		else :
+			/*
+			 * If this file doesn't exist, then we are using the wp-config-sample.php
+			 * file one level up, which is for the develop repo.
+			 */
+			if ( file_exists( ABSPATH . 'wp-config-sample.php' ) ) {
+				$path_to_wp_config = ABSPATH . 'wp-config.php';
+			} else {
+				$path_to_wp_config = dirname( ABSPATH ) . '/wp-config.php';
+			}
 
-		$handle = fopen( $path_to_wp_config, 'w' );
-		foreach ( $config_file as $line ) {
-			fwrite( $handle, $line );
-		}
-		fclose( $handle );
-		chmod( $path_to_wp_config, 0666 );
-		setup_config_display_header();
-		?>
+			$error_message = '';
+			$handle       = fopen( $path_to_wp_config, 'w' );
+			if ( false !== $handle ) {
+				foreach ( $config_file as $line ) {
+					fwrite( $handle, $line );
+				}
+				fclose( $handle );
+			} else {
+				$wp_config_perms = fileperms( $path_to_wp_config );
+				if ( ! empty( $wp_config_perms ) && ! is_writable( $path_to_wp_config ) ) {
+					$wp_config_perms = substr( decoct( $wp_config_perms ), 2 );
+					/* translators: 1: wp-config.php, 2: wp-config.php file permission */
+					$error_message = sprintf(
+							__( 'Unable to write to %1$s file due to its permissions: %2$s. Please try again.' ),
```

Comment:
```suggestion
							__( 'Unable to write to %1$s file due  to too restrictive file permissions: %2$s. Please update the chmod file permissions to allow for write access.' ),
```

Ask in the polyglot channel to validate wording.

Reply:
Code updated in commit https://github.com/WordPress/wordpress-develop/pull/722/commits/c6d272a45e45d97830b04b218b423b0d0a4a6e28